staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo_test.go

```diff
@@ -64,23 +67,35 @@ func TestGetAPIRequestInfo(t *testing.T) {
 		{"GET", "/api/v1/watch/namespaces/other/pods", "watch", "api", "", "v1", "other", "pods", "", "", []string{"pods"}},
 		{"GET", "/api/v1/namespaces/other/pods?watch=1", "watch", "api", "", "v1", "other", "pods", "", "", []string{"pods"}},
 		{"GET", "/api/v1/namespaces/other/pods?watch=0", "list", "api", "", "v1", "other", "pods", "", "", []string{"pods"}},
+		// watch cluster-scoped CRs
+		{"GET", "/apis/stable.example.com/v1/namespaces?watch=true", "watch", "apis", "stable.example.com", "v1", "", "namespaces", "", "", []string{"namespaces"}},
+		{"GET", "/apis/stable.example.com/v1/namespaces/other/namespaces?watch=true", "watch", "apis", "stable.example.com", "v1", "other", "namespaces", "", "", []string{"namespaces"}},
+		// watch namespace-scoped CRs in all namespaces
+		{"GET", "/apis/stable.example.com/v1/namespaces?watch=true", "watch", "apis", "stable.example.com", "v1", namespaceAll, "namespaces", "", "", []string{"namespaces"}},
```

Comment:
Two things: 
- Could you add a comment saying the resource is "namespaces"? otherwise it becomes a little difficult to understand without having the context of this PR and the corresponding issue.
- I may be missing something here but aren't lines 71 and 74 the same verb and URL? I'm guessing the test passes because `namespaceAll` is `""` as well.

Reply:
> * Could you add a comment saying the resource is "namespaces"? otherwise it becomes a little difficult to understand without having the context of this PR and the corresponding issue.

Extended all the test-case comments with more descriptive texts mentioning the API groups appearing in the URLs. Hope they clarify the new test cases now.

> * I may be missing something here but aren't lines 71 and 74 the same verb and URL? I'm guessing the test passes because `namespaceAll` is `""` as well.

Yes, you are right. Test cases for watching cluster-scoped "namespace" resources and watching namespace-scoped "namespace" resources in all namespaces have exactly the same path components. Because they semantically represent different cases (and may be at some point in time `namespaceAll` could differ from `""`), I included them as separate test cases with some explanations as comments. But I think we can also remove one of them.